### DISCUSSION TITLE
Global createMemo shows warning "computations created outside a `createRoot` or `render` will never be disposed"

### DISCUSSION THREAD

**Discussion by ulrichstark** [solidjs/solid#1385] [Q&A] [answered]
Hey guys,

Thanks for your work on SolidJS!
I currently try to migrate an application from React to SolidJS and encountered this warning:

I want to define a global createMemo in its own file, but this leads to the warning "​computations created outside a `createRoot` or `render` will never be disposed".

Do i need to wrap each createMemo in its own createRoot?
Or is there a better solution i am not seeing?

Minimal Project to reproduce: https://playground.solidjs.com/anonymous/9902096e-a539-4be0-a8e9-1a059af7b95f

**Comment by ulrichstark** [comment, marked as answer]
Answered on Solid's Discord: https://discord.com/channels/722131463138705510/1048936508989648986/1049138517286072390

**Reply by yanickrochon** [reply]
Cross-posting. Would've been nice to have the answer, here. The answer is no longer available.

**Reply by ulrichstark** [reply]
@yanickrochon Yes true. I should have shared the answer. I think it was something like "it's not really recommend to have have global createMemo. Rather put them in components and share the memoized values using props or context. If you really want to use a global createMemo, you need to wrap every createMemo individually or all in one createRoot".
